### DISCUSSION TITLE
ckb 0.106.0 (2010923 2022-12-23)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#3771] [Announcements]
### Highlights

**Attention** that this version requires database migration, please backup the data folder and run the following command:

```
ckb migrate
```

This version added better support to estimate the fee rate (#3669). It also has fixed several sync and memory usage bugs. See full changes below.

### Compatability Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2021 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.106.0/ckb_v0.106.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

<details><summary>Changes since v0.105.1</summary>

### Features

* #3669: Add RPC `estimate_cycles` (@zhangsoledad)
* #3703: Tuning the locator algorithm (@driftluo)

### Bug Fixes

* #3675 **light-client:** Subtract overflow when try to get chain root for genesis block (@yangby-cryptape)
* #3697 **light-client:** Do not skip the genesis block for last state proofs (@yangby-cryptape)
* #3698 **tx-pool:** Potential leaks when MaximumAncestors occurs (@zhangsoledad)
* #3706 **tx-pool:** Fix tx-pool potential inconsistent after reorg occurs (@zhangsoledad)
* #3705 **indexer:** Indexer CPU usage (@zhangsoledad)
* #3674: Fix logging feature break (@driftluo)
* #3681: Set recent reject db log file limit to 10 (@driftluo)
* #3699: Resolve a long reorg error (@quake)
* #3702: Unknown list should get header from headermap and db (@driftluo)

### Misc

* #3684: Improve support for cycles access (@zhangsoledad)

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.106.0'>ckb 0.106.0 (2010923 2022-12-23)</a>.</em>

**Comment by joshyates1980** [comment]
For all who choose not to run the ckb node on the C: drive, I use my D: drive
![image](https://user-images.githubusercontent.com/7255545/210153553-1a2096df-1e30-43aa-ae29-888718eb3811.png)

**Comment by joshyates1980** [comment]
Any suggestion on how to fix this error? My main node has not be able to execute for a few weeks now.

![image](https://user-images.githubusercontent.com/7255545/214025970-fd53e637-ff1d-4f53-825f-e2fcf66f2da1.png)

**Reply by eval-exec** [reply]
@joshyates1980 

Hello, what's the output of `ckb --version` ? I suggest you **backup** your data, then try to repair ckb database by `ckb db-repair`.  Or resync ckb node from scratch.
Let me know if they are not working for you.

**Comment by joshyates1980** [comment]
![image](https://user-images.githubusercontent.com/7255545/214174458-3e97883e-b4fe-414d-a0d1-35f6faab8827.png)

**Reply by joshyates1980** [reply]
@eval-exec Any suggestions?  I'm not sure how to troubleshoot this error.
Thanks,
- Josh

**Comment by quake** [comment]
This error appears to be caused by a hardware error in the disk, could you run chkdsk (a windows builtin command) to scan and fix disk first? Then you may need to delete the ckb's data folder to resync the chain
